src/wallet/wallet.h

```diff
@@ -477,21 +477,21 @@ class CWalletTx : public CMerkleTx
 
 class CInputCoin {
 public:
-    CInputCoin()
-    {
-    }
     CInputCoin(const CWalletTx* walletTx, unsigned int i)
     {
-        if (walletTx != nullptr && i < walletTx->tx->vout.size())
+        if (walletTx == nullptr)
+            throw std::invalid_argument("walletTx should not be null");
+        if (i < walletTx->tx->vout.size())

```

Comment:
Ah, I was unclear -- I actually don't mind a `== nullptr` (although current state is cleaner), I was referring to the `throw out_of_range`. It's easier if a constructor is: `if (! preconditionA) throw A; if (!preconditionB) throw B; ....;` rather than `if (! preconditionA) throw A; if (preconditionB)...; else throw B;`